### PR TITLE
Update bundler patches to 1.16.x

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

**Comment by chrisseaton** [comment]
If you search for `1.14.6` you'll find we document this version in several other places. Do they need to be updated as well?

Did you check that the licence hasn't changed in any way? Including copyright years?

**Comment by pitr-ch** [comment]
I've updated the licenses format so it matches exactly, and the documentation.

File: src/main/java/org/truffleruby/parser/BodyTranslator.java [hunk 1]
```diff
@@ -2450,51 +2450,84 @@ public RubyNode visitOpElementAsgnNode(OpElementAsgnParseNode node) {
...
+                        node.getPosition(),
+                        read,
+                        node.getOperatorName(),
+                        buildArrayNode(node.getPosition(), right),
+                        null);
+                break;
+        }
+
+        copyNewline(node, operation);
+        return operation;
```
Comment (eregon) [inline src/main/java/org/truffleruby/parser/BodyTranslator.java]: This logic needs a spec or removing a tag if there is already a spec.

Comment (pitr-ch) [reply src/main/java/org/truffleruby/parser/BodyTranslator.java]: I am making a note and I'll check or add later. I need this merged asap for 33

File: lib/patches/bundler/bundler/deprecate.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+begin
+  require "rubygems/deprecate"
+rescue LoadError
+  # it's fine if it doesn't exist on the current RubyGems...
+  nil
+end
+
+module Bundler
+  # TODO (pitr-ch 26-Mar-2018): we need to also check that it's not autoload cosntant
```
Comment (eregon) [inline lib/patches/bundler/bundler/deprecate.rb]: It's MRI semantics to have it defined if it's an autoload:
```ruby
pry
[1] pry(main)> autoload :Foo, "foo.rb"
=> nil
[2] pry(main)> defined? Foo
=> "constant"
```

Comment (pitr-ch) [reply lib/patches/bundler/bundler/deprecate.rb]: Hm then the problem is somewhere else then. It fixes the immediate problem with bundler nevertheless.

Comment (eregon) [reply lib/patches/bundler/bundler/deprecate.rb]: Also:
```ruby
[1] pry(main)> module Bundler
[1] pry(main)*   autoload :Deprecate, "depr"
[1] pry(main)* end  
=> nil
[2] pry(main)> defined?(Bundler::Deprecate) 
=> "constant"
```

**Pull request merged**